src/storage/src/read/windowed.rs

```diff
@@ -60,6 +62,7 @@ where
     R: BatchReader,
 {
     async fn next_batch(&mut self) -> Result<Option<Batch>> {
+        let _window_scan_elapsed = timer!(crate::metrics::WINDOW_SCAN_ELAPSED);
```

Comment:
This only records elapsed time for a `next_batch()` call, not the whole scan. We can record the elapsed time on `drop()` to get the scan time.

Reply:
This is intentially left to monitor per-window cost time.